debug/targets.py

```diff
@@ -180,5 +180,10 @@ def target(parsed):
 
     t = found[0](parsed.target, parsed)
     assert t.harts, "%s doesn't have any harts defined!" % t.name
+    for h in t.harts :
+        if (h.xlen == 0):
+            h.xlen = parsed.xlen
+        elif (h.xlen != parsed.xlen):
+            raise Exception("Hart had a specified XLEN of %d and target had a specified XLEN of %d, they must match." % (h.xlen, parsed.xlen))
```

Comment:
Originally this option would allow you to override XLEN as well. Given that changing XLEN often requires a new link map I don't know how useful that really was, so I'm fine with this new requirement.

I'd say this instead:
"The target has an XLEN of %d, but the command line specified an XLEN of %d. They must match."